docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt

```diff
@@ -51,16 +49,176 @@ Resource budget:
+by the adaptive algorithm. Note that although the name "ASHA" includes
+the phrase "halving", the fraction of trials pruned after every rung is
+controlled by ``divisor``.
+
+Adaptive over ASHA
+==================
+
+The adaptive algorithm calls ASHA subroutines with varying parameters.
+The exact calls are configured through the choice of ``mode``, which
+specifies how aggressively to perform early stopping. One way to think
+about this behavior is as a spectrum that ranges from "one ASHA run"
+(aggressive early stopping; eliminate most trials every rung) to
+"``searcher: random``" "multiple ASHA runs, some of which will not early
+stop and others will early stop later" (try some without early stopping;
+initialized trials may be allowed to run to completion).
+
+On one end, ``aggressive`` applies early stopping in a very eager
+manner; this mode essentially corresponds to only making a single call
+to ASHA. With the default ``divisor`` of 4, 75% of the remaining trials
+will be eliminated in each rung after only being trained for 25% the
+length of the next rung. This implies that relatively few trials will be
+allowed to finish even a small fraction of the length needed train to
+convergence (``max_length``). This aggressive early stopping behavior
+allows the searcher to start more trials for a wider exploration of
+hyperparameter configurations, at the risk of discarding a configuration
+too soon.
+
+On the other end, ``conservative`` mode is more similar to a ``random``
+search, in that it performs significantly less pruning. Extra ASHA
+subroutines are spawned with fewer rungs and longer training lengths to
+account for the high percentage of trials eliminated after only a short
+time. However, a ``conservative`` adaptive search will only explore a
+small fraction of the configurations explored by an ``aggressive``
+search, given the same budget.
+
+Once the number and types of calls to ASHA are determined (via
+``mode``), the adaptive algorithm will allocate training length budgets
+to the ASHA subroutines, from the overall budget for the adaptive
+algorithm (user-specified through ``budget``). This determines the
+number of trials at each rung (N in the above ASHA example).
+
+Why Asynchronous Halving?
+=========================
+
+Synchronous halving promotes hyperparameter configurations
+synchronously, waiting for each rung to complete before performing any
+promotions. This allows the algorithm to have complete information about
+all trials at the time of promotion, but it results in underutilized
+nodes waiting on completion of validation steps for other
+configurations. ASHA, or asynchronous halving, uses asynchronously
+promotes trials when it has the minimum information required to make a
+decision in order to maximize compute efficiency of the searcher.
 
 See the difference in asynchronous vs. sychronous promotions in the two
 animated GIFs below:
 
 .. image:: /assets/images/sha.gif
 
 .. image:: /assets/images/asha.gif
+
```

Comment:
Please add:
In contrast to SHA which initializes all trials in the bottom rung at the outset, ASHA will continuously add trials to the bottom rung until the desired number of trials is reached.